### PR TITLE
Fixed support for physical keyboards on Android. flutter/flutter#11177

### PR DESCRIPTION
**Pull request by slightfoot**
Fixes flutter/flutter#11177

Also fixes flutter/flutter#7943 and fixes flutter/flutter#9347

> @slightfoot **Waits patiently for a merge.**
> I can't wait to be able to type things into TextField from my actual keyboard rather than using the silly on-screen soft keyboard.. such a pain.

### CODE REVIEW

**Comment by slightfoot** [comment]
@xster ping!. the original bug was not assigned, not sure who can review this PR.

**Comment by xster** [comment]
@cbracken or @jason-simmons would be more familiar with the text editing interactions.

File: shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java [hunk 1]
```diff
@@ -128,36 +128,47 @@ public boolean setSelection(int start, int end) {
...
-        final boolean result = super.sendKeyEvent(event);
-        if (event.getAction() == KeyEvent.ACTION_UP) {
+        if (event.getAction() == KeyEvent.ACTION_DOWN) {
             if (event.getKeyCode() == KeyEvent.KEYCODE_DEL) {
                 int selStart = Selection.getSelectionStart(mEditable);
                 int selEnd = Selection.getSelectionEnd(mEditable);
                 if (selEnd > selStart) {
                     // Delete the selection.
                     Selection.setSelection(mEditable, selStart);
-                    deleteSurroundingText(0, selEnd - selStart);
```
Comment (cbracken) [inline shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java]: Looks like you can delete `deleteSurroundingText` now. I don't see any other uses.

Comment (cbracken) [reply shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java]: Apologies - ignore that; it's part of the public interface; leave it for now and we can take a look at cleaning up separately.

Comment (slightfoot) [reply shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java]: sounds good, anything else on the review?

**Comment by cbracken** [comment]
> Googler, please make sure the reason for overriding the CLA status is clearly documented in these comments.

@slightfoot verified to have signed CLA. @cbracken is a Googler.

**Comment by cbracken** [comment]
LGTM. Post-fixes. Manually verified for both English and Japanese input on a Nexus 6P. Thanks for sending!

lgtm

**Comment by slightfoot** [comment]
@cbracken thanks! great news!

**Pull request merged**